libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -790,10 +807,65 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 			_assembly.appendInstruction(Instruction::POP);
 		}
 	};
-	solAssert(_inlineAssembly.annotation().analysisInfo, "");
+
+	yul::Block const* code = &_inlineAssembly.operations();
+	yul::AsmAnalysisInfo* analysisInfo = &*_inlineAssembly.annotation().analysisInfo;
+
+	yul::Object object;
```

Comment:
Only needed inside the if.

Reply:
If i move it inside the scope of the if, the shared pointers would be deallocated